tests/data/Compressor.cpp

```diff
@@ -0,0 +1,251 @@
+    }
+
+    // evaluate parsed arguments
+    if( vm.count( "help" ))
+    {
+        std::cout << options << std::endl;
+        ::exit( EXIT_SUCCESS );
+    }
+
+    Strings files;
+    if( vm.count( "data" ))
+        files = vm["data"].as< Strings >();
+    else
+    {
+        getFiles( files, ".*\\.dll" );
+        getFiles( files, ".*\\.exe" );
+        getFiles( files, ".*\\.so" );
+        getFiles( files, ".*\\.a" );
+        getFiles( files, ".*\\.dylib" );
+        getFiles( files, ".*\\.rgb" );
+        getFiles( files, ".*\\.bin" );
+        getFiles( files, ".*\\.ply" );
+        getFiles( files, ".*\\.bbp" );
+
+        // Limit to 30 files using a pseudo-random selection for reproducability
+        const size_t maxFiles = 30;
+        if( files.size() > maxFiles )
+        {
+            const size_t cut = files.size() - maxFiles;
+            for( size_t i = 0; i < cut; ++i )
+                files.erase( files.begin() +
+                             (i * 997 /*prime*/) % files.size( ));
+        }
+    }
+
+    std::cout.setf( std::ios::right, std::ios::adjustfield );
+    std::cout.precision( 5 );
+    std::cout << "                File, Compressor, Uncompress, "
+              << "Compressed,   comp MB/s, decomp MB/s" << std::endl;
+    const auto& infos = getCompressors();
+    for( const auto& info : infos )
+    {
+        _result = 0;
+        _size = 0;
+        _compressionTime = 0;
+        _decompressionTime = 0;
+
+        for( const auto& file : files )
+        {
+            lunchbox::MemoryMap map( file );
+            const uint8_t* data = map.getAddress< const uint8_t >();
+            const size_t size = map.getSize();
+
+            if( !data || size > LB_1GB )
+                continue;
+
+            const std::string name = lunchbox::getFilename( file );
+            _testData( info, name, data, size );
+        }
+        if( _baseTime == 0.f )
+            _baseTime = _compressionTime + _decompressionTime;
+
+        std::cout << std::setw(24) << "Total, " << info.name
+                  << std::setfill(' ') << ", " << std::setw(10) << _size << ", "
+                  << std::setw(10) << _result << ", " << std::setw(10)
+                  << float(_size) / 1024.f / 1024.f * 1000.f / _compressionTime
+                  << ", " << std::setw(10)
+                  << float(_size) / 1024.f / 1024.f * 1000.f /_decompressionTime
+                  << std::endl
+                  << "    info->ratio = " << float(_result) / float(_size)
+                  << "f;" << std::endl
+                  << "    info->speed = " << float(_baseTime) /
+                                    float(_compressionTime + _decompressionTime)
+                  << "f;" << std::endl << std::endl;
+    }
+}
+
+void _testRandom()
+{
+    ssize_t size = LB_10MB;
+    uint8_t* data = new uint8_t[size];
+    lunchbox::RNG rng;
+#pragma omp parallel for
+    for( ssize_t i = 0; i < size; ++i )
+        data[i] = rng.get< uint8_t >();
```

Comment:
Not that it matters, but I think calling rng in parallel can skew the distribution. A was told about a similar effect when consistently discarding 1 out of every x numbers.